code/controllers/subsystem/vote.dm

```diff
@@ -6,42 +6,36 @@ SUBSYSTEM_DEF(vote)
 
 	runlevels = RUNLEVEL_LOBBY | RUNLEVELS_DEFAULT
 
-	var/initiator = null
-	var/started_time = null
+	var/mode
+	var/question
+	var/initiator
+	var/started_time
 	var/time_remaining = 0
-	var/mode = null
-	var/question = null
-	var/list/choices = list()
 	var/list/voted = list()
 	var/list/voting = list()
+	var/list/choices = list()
+	var/list/choice_by_ckey = list()
 	var/list/generated_actions = list()
 
-/datum/controller/subsystem/vote/fire() //called by master_controller
-	if(mode)
-		time_remaining = round((started_time + CONFIG_GET(number/vote_period) - world.time)/10)
-
-		if(time_remaining < 0)
-			result()
-			for(var/client/C in voting)
-				C << browse(null, "window=vote;can_close=0")
-			reset()
-		else
-			var/datum/browser/client_popup
-			for(var/client/C in voting)
-				client_popup = new(C, "vote", "Voting Panel")
-				client_popup.set_window_options("can_close=0")
-				client_popup.set_content(interface(C))
-				client_popup.open(FALSE)
-
+/datum/controller/subsystem/vote/fire()	//called by master_controller
+	if(!mode)
+		return
+	time_remaining = round((started_time + CONFIG_GET(number/vote_period) - world.time)/10)
```

Comment:
This has no reason to be a countdown constantly processed by fire. Make it a timer.

Reply:
removing this block

Reply:
See previous comment on time_remaining